File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    unordered_set<int>us;
    bool go(TreeNode *root,int k){
        if(root==NULL) return false;
        if(us.find(k-root->val)!=us.end()) return true;
        us.insert(root->val);
        return (go(root->left,k) or go(root->right,k));
    }
    bool findTarget(TreeNode* root, int k) {
        return go(root,k);
    }
};

File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        dic  =[]
        def go(root):
            if root==None:
                return False
            if (k-root.val) in dic :
                return True
            dic.append(root.val)
            return go(root.left) or go(root.right)
        return go(root)